src/clipboardobserver.js

```diff
@@ -22,17 +22,53 @@ export default class ClipboardObserver extends DomEventObserver {
 	constructor( doc ) {
 		super( doc );
 
-		this.domEventType = [ 'paste', 'copy', 'cut' ];
+		this.domEventType = [ 'paste', 'copy', 'cut', 'drop' ];
+
+		doc.on( 'paste', _handleInput, { priority: 'low' } );
+		doc.on( 'drop', _handleInput, { priority: 'low' } );
+
+		function _handleInput( evt, data ) {
+			// Prevent page refreshing.
+			data.preventDefault();
+
+			doc.fire( 'input', { dataTransfer: data.dataTransfer } );
```

Comment:
This event is really confusing. There's a native`input` event. So this one would need to be called `clipboardInput`. Is it ok?